### ISSUE TITLE
Ensure buttons have discernible text (.common-button[role="button"]): A11y_Microsoft MakeCOde_Settings_Extensions_AI4W

### ISSUE THREAD

**Issue by kupatkar99** [microsoft/pxt-microbit#6831]
**"Leveraging SWE Agent can help accelerate this accessibility bug fix; review and validate all AI generated changes before merge."** 

**"Try [ES Chat](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2feng.ms%2fdocs%2fcoreai%2fdevdiv%2fone-engineering-system-1es%2f1es-docs%2fes-chat%2findex&data=05%7c02%7cv-mchougule%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838215421556%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=%2B1yeHqbNhHcMuQaZNXDo%2B%2fiWxH0mZFSmWpamQk2ftws%3D&reserved=0) to learn more about the MAS rule and how to fix the issue. If you need more help, use our [Teams channel](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2fteams.microsoft.com%2fl%2fchannel%2f19%253ADeCtYeREyQHw5jYsz9PPc3hpVw48m-tOlTkQ1jEn0eM1%2540thread.tacv2%2fCAI%2520Accessibility%2520Support%2520Channel?groupId%3Df5eb428c-9e6e-4c16-8143-33fe3961a9ae%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=05%7c02%7cv-mchougule%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838215450064%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=yncZXSKXaJlQ3HpXeR3TK03MNIhUe8tV2yZl49clFZY%3D&reserved=0) or [office hours](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2foutlook.office.com%2fbook%2fmicrosoft2779007%2540microsoft.onmicrosoft.com%2fs%2fbLeGBltMW0qI0WRLsTqx2A2?ismsaljsauthenabled&data=05%7c02%7cv-mchougule%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838215467554%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=i4Fn6qPJw4eg3Ny6En0%2fPnIqw%2BpJpHuT%2f%2BZH2hE%2fRTA%3D&reserved=0)."**

"[Check out Accessibility Insights](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0)! - Identify accessibility bugs before check-in and make bug fixing faster and easier."

"Contact [CAI Accessibility Support Channel](https://teams.microsoft.com/l/channel/19:DeCtYeREyQHw5jYsz9PPc3hpVw48m-tOlTkQ1jEn0eM1%40thread.tacv2/CAI%20Accessibility%20Support%20Channel?groupId=f5eb428c-9e6e-4c16-8143-33fe3961a9ae&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) for Accessibility queries on design, development, test, usability or any processes"

**Please do not close this bug. This bug should only be closed by Trusted Tester after verification.**

### GitHub Tags:
#A11yTCS; #A11yMAS; #A11yAuto; #Win11; #DesktopWeb; #Microsoft MakeCode; #E2E_MicrosoftMakeCode_Web_May2026; #FTP; #ChromiumEdge; #Name Role Value; #A11ySev2; #WCAG4.1.2; #AI4W; #ES Chat; #copilot:agent=a11yagent; #button-name;

### Environment Details:
Application Name: Microsoft MakeCode
URL: https://makecode.microbit.org/beta
windows Edition: Windows 11 Enterprise Insider Preview(24H2)
OS Build: 26544.1000

### Repro Steps:
1. Launch application URL : https://makecode.microbit.org
2. Home page will open.
3. Tab to "New Project" option and select it. Editor screen will open.
5. Tab to "Settings" option appearing in header section and press enter key.
6. Tab to "Extensions" option and press enter key.
7. Extension page will open.
8. Run accessibility insights for web.
9. Observe the issue.

### Issue:
Ensure buttons have discernible text (button-name - https://accessibilityinsights.io/info-examples/web/button-name)

### Target application:
 Microsoft MakeCode for micro:bit - https://makecode.microbit.org/beta#editor

### Element path:
 .common-input-group > .common-button[role="button"]

### Snippet:
<button class="common-button" role="button" tabindex="0"><span class="common-button-flex"><i class="fas fa-search" aria-hidden="true"></i><span class="common-button-label"></span></span></button>

### How to fix: 
Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element does not have an implicit (wrapped) <label>
  Element does not have an explicit <label>
  Element's default semantics were not overridden with role="none" or role="presentation"

**Environment:** Microsoft Edge version 148.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.46.0 (axe-core 4.10.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

### Attachment:

<img width="959" height="476" alt="Image" src="https://github.com/user-attachments/assets/795e50d6-56a6-4c67-8468-d7b34c8fd9d8" />

**Comment by riknoll** [comment]
fixed by pxt#11410
